lago/subnet_lease.py

```
@@ -17,222 +17,505 @@
+        octet.
+
+        Args:
+            idx (str): The value of the third octet
+
+        Returns:
+            Lease: Lease object which represents the requested subnet.
+
+        Raises:
+            LagoSubnetLeaseOutOfRangeException: If the resultant subnet is
+            malformed or out of the range of the store.
+        """
+
+        return self.create_lease_object_from_subnet(
+            self._subnet_template.format(idx)
+        )
+
+    def create_lease_object_from_subnet(self, subnet):
+        """
+        Create a lease from ip in a dotted decimal format,
+        (for example '192.168.200.0/24'). the _cidr will be added if not exist
+        in 'subnet'.
+
+        Args:
+            subnet (str): The value of the third octet
+
+        Returns:
+            Lease: Lease object which represents the requested subnet.
+
+        Raises:
+            LagoSubnetLeaseOutOfRangeException: If the resultant subnet is
+            malformed or out of the range of the store.
+        """
+        if '/' not in subnet:
+            subnet = '{}/{}'.format(subnet, self._cidr)
+        if not self.is_leasable_subnet(subnet):
+            raise LagoSubnetLeaseOutOfRangeException(
+                subnet, self.get_allowed_range()
+            )
+
+        return Lease(store_path=self.path, subnet=subnet)
+
+    def is_leasable_subnet(self, subnet):
+        """
+        Checks if a given subnet is inside the defined provision-able range
+
+        Args:
+            subnet (str): Ip in dotted decimal format with _cidr notation
+                (for example '192.168.200.0/24')
+
+        Returns:
+            bool: True if subnet can be parsed into IPNetwork object and is
+                inside the range, False otherwise
+        """
+        try:
+            return \
+                self._min_subnet <= \
+                IPNetwork(subnet) <= \
+                self._max_subnet
+        except AddrFormatError:
```

Doesn't this mask the error if the subnet format was given improperly?